crates/subspace-service/src/root_blocks.rs

```diff
@@ -0,0 +1,80 @@
+use futures::{Stream, StreamExt};
+use parity_scale_codec::{Decode, Encode};
+use sc_client_api::backend::AuxStore;
+use sc_consensus_subspace::ArchivedSegmentNotification;
+use std::error::Error;
+use std::sync::Arc;
+use subspace_core_primitives::{RootBlock, SegmentIndex};
+use tracing::{debug, error, trace};
+
+/// Start an archiver that will listen for archived segments and send root block to the storage
+pub(crate) async fn start_root_block_archiver<AS: AuxStore>(
+    mut root_block_cache: RootBlockCache<AS>,
+    mut archived_segment_notification_stream: impl Stream<Item = ArchivedSegmentNotification> + Unpin,
+) {
+    trace!("Subspace root block archiver started.");
+
+    while let Some(ArchivedSegmentNotification {
+        archived_segment, ..
+    }) = archived_segment_notification_stream.next().await
+    {
+        let segment_index = archived_segment.root_block.segment_index();
+        let result = root_block_cache.add_root_block(archived_segment.root_block);
+
+        if let Err(err) = result {
+            error!(%segment_index, ?err, "Root block archiving failed.");
+        } else {
+            debug!(%segment_index, "Root block archived.");
+        }
+    }
+}
+
+/// Cache of recently produced root blocks in aux storage
+pub struct RootBlockCache<AS> {
+    aux_store: Arc<AS>,
+}
+
+impl<AS> RootBlockCache<AS>
+where
+    AS: AuxStore,
+{
+    const KEY_PREFIX: &[u8] = b"root-block-cache";
```

Comment:
```suggestion
    const KEY_PREFIX: &[u8] = b"segment-headers-cache";
```

Since these keys are hard to change later, we should prefer long-term name even though root blocks are otherwise not refactored to segment headers yet.